receiver/hostmetricsreceiver/internal/scraper/obsreportscraper/obsreportscraper_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package obsreportscraper
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opencensus.io/trace"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+func TestWrapScraper(t *testing.T) {
+	ts := &testScraper{
+		t:   t,
+		err: nil,
+	}
+	obss := WrapScraper(ts, "test")
+	assert.NoError(t, obss.Initialize(context.Background()))
+	assert.NoError(t, obss.Close(context.Background()))
```

Comment:
It doesn't matter for this test but this should really be deferred (or at least after the ScrapeMetrics line)

Likewise for the other tests

Reply:
Done. I want to check the result and defer does not help so moved as a last thing.